saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/sync/SavedSitesSyncDataObserver.kt

```diff
@@ -38,28 +41,43 @@ import timber.log.Timber
 class SavedSitesSyncDataObserver @Inject constructor(
     private val savedSitesRepository: SavedSitesRepository,
     private val syncEngine: SyncEngine,
-    private val deviceSyncState: DeviceSyncState,
+    private val syncStateMonitor: SyncStateMonitor,
     @AppCoroutineScope private val coroutineScope: CoroutineScope,
-    private val dispatcherProvider: DispatcherProvider,
 ) : MainProcessLifecycleObserver {
 
+    private var dataObserverJob: Job? = null
     private var initialised: Boolean = false
 
     override fun onCreate(owner: LifecycleOwner) {
         super.onCreate(owner)
-        if (deviceSyncState.isUserSignedInOnDevice()) {
-            Timber.d("Sync-Feature: Listening to changes in Saved Sites")
-            coroutineScope.launch(dispatcherProvider.io()) {
-                savedSitesRepository.lastModified().collect {
-                    if (initialised) {
+        syncStateMonitor.syncState()
+            .onEach { state ->
+                when (state) {
+                    OFF -> cancelSavedSitesChanges()
+                    else -> observeSavedSitesChanges()
+                }
+            }
+            .launchIn(coroutineScope)
+    }
+
+    private fun observeSavedSitesChanges() {
+        if (dataObserverJob == null) {
+            dataObserverJob = coroutineScope.launch {
+                if (initialised) {
```

Comment:
`savedSitesRepository.lastModified()` will immediately return a value, and that will trigger a call to sync. Since we also have `AppLifecycleSyncObserver` that triggers Sync on app start, this flag avoid duplication. There might be better ways.

Reply:
I see, if that's a problem I would try to handle that in the internal logic and not from the caller.
My proposal: let's remove it and if at some point that's a problem we can think about possible solutions. Actually, I think if both are triggered at the same time, one should hit "sync in progress" and will be discarded.

Reply:
I’m not sure I agree, the trigger is different. AFAIK (if you know how please share it) we can’t force the order in what the `MainProcessLifecycleObserver` impl are called. So if `SavedSitesSyncDataObserver` is called before the `AppLifecycleSyncObserver`, the sync that will be trigger will be `DATA_CHANGE` and not `APP_OPEN`. 

Just trying to be consistent with the flows that were described.

Reply:
Resolving, we discussed this one on zoom 👍